src/main/kotlin/com/thelastpickle/tlpstress/converters/HumanReadableTimeConverter.kt

```diff
@@ -1,40 +1,38 @@
 package com.thelastpickle.tlpstress.converters
 
 import com.beust.jcommander.IStringConverter
-import org.joda.time.Period
-import org.joda.time.format.PeriodFormatterBuilder
-import org.joda.time.format.PeriodFormatter
-
-
-
-class HumanReadableTimeConverter : IStringConverter<Int> {
-    override fun convert(value: String?): Int {
-        val daysFormatter = PeriodFormatterBuilder()
-                .appendDays().appendSuffix("d").appendSeparatorIfFieldsAfter(" ")
-                .appendHours().appendSuffix("h").appendSeparatorIfFieldsAfter(" ")
-                .appendMinutes().appendSuffix("m")
-                .toFormatter();
-
-        val hoursFormatter = PeriodFormatterBuilder()
-                .appendHours().appendSuffix("h").appendSeparatorIfFieldsAfter(" ")
-                .appendMinutes().appendSuffix("m")
-                .toFormatter();
-
-        val minutesFormatter = PeriodFormatterBuilder()
-                .appendMinutes().appendSuffix("m")
-                .toFormatter();
-
-
-        var duration = Period()
-
-        if (value!!.contains('d')) {
-            duration = daysFormatter.parsePeriod(value)
-        } else if (value!!.contains('h')) {
-            duration = hoursFormatter.parsePeriod(value)
-        } else {
-            duration = minutesFormatter.parsePeriod(value)
-        }
-
-        return duration.toStandardMinutes().minutes
+import java.time.Duration
+
+
+class HumanReadableTimeConverter : IStringConverter<Long> {
+    override fun convert(value: String?): Long {
+        var duration = Duration.ofMinutes(0)
+
+        val integer = "\\d+"
+        val units = "[dhms]"
+        val valueCharSequence = value!!.subSequence(0, value.length)
+        /**
+         * The duration is passed in via the value variable. It could contain multiple time values e.g. "1d 2h 3m 4s".
+         * Parse the string using the following process:
+         * 1. Find all occurrences of of an integer with a time unit and iterate through the matches. Note we need
+         *      to convert the value from a String to CharSequence so we can pass it to findAll.
+         * 2. Iterate through the matched values. Add to the duration based on the units of each value.
+         */
+        Regex("$integer$units")
+            .findAll(valueCharSequence)
+            .forEach {
+                val quantity = Regex(integer).findAll(it.value).first().value.toLong()
+                when (Regex(units).findAll(it.value).first().value) {
```

Comment:
This is a super minor nit, feel free to put it in a follow up issue if you don't feel like doing it now.  Using capture groups around the original regex will let you avoid needing to do a second one.  Example: `((\d+)s?([dmhs]))+`

The capture groups can be looped over and you don't need to evaluate the second regex.  It also cuts down on the complexity of the code, you can skip the `$integer` and `$units` string evaluation.

Reply:
I agree. The current method is a bit clunky. It should be pretty quick to fix.